From 6d6ec9e80633fb24308061c4be0cc64e3c0bc191 Mon Sep 17 00:00:00 2001
From: Ting Liu <ting.liu@nxp.com>
Date: Fri, 9 Oct 2020 21:15:34 +0530
Subject: [PATCH] add typedef to avoid multiple definition issue

powerpc-fsl-linux-ld: board/freescale/t102xrdb/cpld.o:board/freescale/t102xrdb/cpld.h:24: multiple definition of `cpld_data_t'; board/freescale/t102xrdb/t102xrdb.o:board/freescale/t102xrdb/cpld.h:24: first defined here
.../scripts/Makefile.build:355: recipe for target 'board/freescale/t102xrdb/built-in.o' failed
make[2]: *** [board/freescale/t102xrdb/built-in.o] Error 1

Signed-off-by: Ting Liu <ting.liu@nxp.com>
---
 board/freescale/t102xrdb/cpld.h | 2 +-
 board/freescale/t104xrdb/cpld.h | 2 +-
 board/freescale/t208xrdb/cpld.h | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/board/freescale/t102xrdb/cpld.h b/board/freescale/t102xrdb/cpld.h
index c05f536806..48dd37a775 100644
--- a/board/freescale/t102xrdb/cpld.h
+++ b/board/freescale/t102xrdb/cpld.h
@@ -4,7 +4,7 @@
  *
  */
 
-struct cpld_data {
+typedef struct cpld_data {
 	u8 cpld_ver;		/* 0x00 - CPLD Major Revision Register */
 	u8 cpld_ver_sub;	/* 0x01 - CPLD Minor Revision Register */
 	u8 hw_ver;		/* 0x02 - Hardware Revision Register */
diff --git a/board/freescale/t104xrdb/cpld.h b/board/freescale/t104xrdb/cpld.h
index a816aef10a..5fc9fd9c2d 100644
--- a/board/freescale/t104xrdb/cpld.h
+++ b/board/freescale/t104xrdb/cpld.h
@@ -9,7 +9,7 @@
 /*
  * CPLD register set. Feel free to add board-specific #ifdefs where necessary.
  */
-struct cpld_data {
+typedef struct cpld_data {
 	u8 cpld_ver;		/* 0x00 - CPLD Major Revision Register */
 	u8 cpld_ver_sub;	/* 0x01 - CPLD Minor Revision Register */
 	u8 hw_ver;		/* 0x02 - Hardware Revision Register */
diff --git a/board/freescale/t208xrdb/cpld.h b/board/freescale/t208xrdb/cpld.h
index bd6c203742..7cb2513f18 100644
--- a/board/freescale/t208xrdb/cpld.h
+++ b/board/freescale/t208xrdb/cpld.h
@@ -6,7 +6,7 @@
 /*
  * CPLD register set of T2080RDB board-specific.
  */
-struct cpld_data {
+typedef struct cpld_data {
 	u8 chip_id1;		/* 0x00 - Chip ID1 register */
 	u8 chip_id2;		/* 0x01 - Chip ID2 register */
 	u8 hw_ver;		/* 0x02 - Hardware Revision Register */
-- 
2.17.1

